scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java

```diff
@@ -389,23 +411,27 @@ static TaskData createTaskData(JobData jobRuntimeData, InternalTask task) {
         taskData.updateMutableAttributes(task);
 
         if (task.getSelectionScripts() != null) {
-            List<ScriptData> scripts = new ArrayList<>(task.getSelectionScripts().size());
+            List<SelectionScriptData> scripts = new ArrayList<>(task.getSelectionScripts().size());
             for (SelectionScript selectionScript : task.getSelectionScripts()) {
-                scripts.add(ScriptData.createForSelectionScript(selectionScript, taskData));
+                scripts.add(SelectionScriptData.createForSelectionScript(selectionScript, taskData));
             }
             taskData.setSelectionScripts(scripts);
         }
+        if (task.getExecutableContainer() != null) {
+            taskData.setScript(ScriptData.createForScript(
+                    ((ScriptExecutableContainer) task.getExecutableContainer()).getScript(), taskData));
+        }
         if (task.getPreScript() != null) {
-            taskData.setPreScript(ScriptData.createForScript(task.getPreScript()));
+            taskData.setPreScript(ScriptData.createForScript(task.getPreScript(), taskData));
```

Comment:
Please put `task.getX` in a variable in order to factorize  `taskData.setPreScript(ScriptData.createForScript(...)`